core/types/transaction.go

```diff
@@ -573,13 +570,18 @@ func (tx *Transaction) WithSignature(signer Signer, sig []byte) (*Transaction, e
 	return &Transaction{inner: cpy, time: tx.time}, nil
 }
 
-// IntrinsicGas computes the 'intrinsic gas' for a message with the given data.
+// IntrinsicGas returns the 'intrinsic gas' computed for a message with the given data.
 func (tx *Transaction) IntrinsicGas(rules *params.Rules) (uint64, error) {
+	return IntrinsicGas(tx.inner, rules)
+}
+
+// IntrinsicGas computes the 'intrinsic gas' for a message with the given data.
+func IntrinsicGas(txdata TxData, rules *params.Rules) (uint64, error) {
```

Comment:
It's annoying to have this function return an error. The only cases where it could return an error is for very large data, which is highly unlikely in practice. It should panic in these cases instead.

The function `IntrinsicGas` is supposed to be used when construction a new transaction. If we want to be even more safe, we could keep the error-returning variant as an internal function for use in `Transaction.IntrinsicGas`, since that one is used to validate transactions received over p2p.

Reply:
I did exactly this 👍